app/scripts/modules/core/src/pipeline/config/triggers/wercker/WerckerTriggerTemplate.tsx

```diff
@@ -0,0 +1,158 @@
+import * as React from 'react';
+import { Option } from 'react-select';
+import { $q } from 'ngimport';
+import { IPromise } from 'angular';
+
+import { IBuild, IBuildInfo, IBuildTrigger } from 'core/domain';
+import { ITriggerTemplateComponentProps } from 'core/pipeline/manualExecution/TriggerTemplate';
+import { IgorService } from 'core/ci';
+import { Spinner } from 'core/widgets/spinners/Spinner';
+import { buildDisplayName } from 'core/pipeline/executionBuild/buildDisplayName.filter';
+import { timestamp } from 'core/utils/timeFormatters';
+import { TetheredSelect } from 'core/presentation/TetheredSelect';
+
+export interface IWerckerTriggerTemplateState {
+  builds?: IBuild[];
+  buildsLoading?: boolean;
+  loadError?: boolean;
+  selectedBuild?: number;
+}
+
+export class WerckerTriggerTemplate extends React.Component<
+  ITriggerTemplateComponentProps,
+  IWerckerTriggerTemplateState
+> {
+  public static formatLabel(trigger: IBuildTrigger): IPromise<string> {
+    return $q.when(`(Wercker) ${trigger.master}: ${trigger.job}`);
+  }
+
+  public constructor(props: ITriggerTemplateComponentProps) {
+    super(props);
+    this.state = {
+      builds: [],
+      buildsLoading: false,
+      loadError: false,
+      selectedBuild: 0,
+    };
+  }
+
+  private buildLoadSuccess = (allBuilds: IBuild[]) => {
+    const newState: Partial<IWerckerTriggerTemplateState> = {
+      buildsLoading: false,
+    };
+
+    const trigger = this.props.command.trigger as IBuildTrigger;
+    newState.builds = allBuilds
+      .filter(build => !build.building && build.result === 'SUCCESS')
+      .sort((a, b) => b.number - a.number);
+    if (newState.builds.length) {
+      // default to what is supplied by the trigger if possible; otherwise, use the latest
+      const defaultSelection = newState.builds.find(b => b.number === trigger.buildNumber) || newState.builds[0];
+      newState.selectedBuild = defaultSelection.number;
+      this.updateSelectedBuild(defaultSelection);
+    }
+
+    this.setState(newState);
+  };
+
+  private buildLoadFailure = () => {
+    this.setState({
+      buildsLoading: false,
+      loadError: true,
+    });
+  };
+
+  private updateSelectedBuild = (item: any) => {
+    this.props.command.extraFields.buildNumber = item.number;
+    this.setState({ selectedBuild: item.number });
+  };
+
+  private initialize = () => {
+    const { command } = this.props;
+    const trigger = command.trigger as IBuildTrigger;
+
+    // These fields will be added to the trigger when the form is submitted
+    command.extraFields = {};
+
+    this.setState({
+      buildsLoading: true,
+      loadError: false,
+    });
+
+    if (trigger.buildNumber) {
+      this.updateSelectedBuild(trigger.buildInfo);
+    }
+
+    // do not re-initialize if the trigger has changed to some other type
+    if (trigger.type !== 'wercker') {
```

Comment:
Check equality against `BuildServiceType.Wercker` enum value here instead of string literal `'wercker'`.  Add `BuildServiceType` to import on line 8.